doc/development_guide/how_tos/custom_checkers.rst

```diff
@@ -241,6 +218,55 @@ Now we can debug our checker!
     environment variable or by adding the ``my_plugin.py``
     file to the ``pylint/checkers`` directory if running from source.
 
+Defining a Message
+------------------
+
+Pylint message is defined using the following format::
+
+   msgs = {
+       "message-id": (
+           "displayed-message", "message-symbol", "message-help"
+       )
+   }
+
+
+* The ``message-id`` should be a 4-digit number,
+  prefixed with a **message category**.
+  There are multiple message categories,
+  these being ``C``, ``W``, ``E``, ``F``, ``R``,
+  standing for ``Convention``, ``Warning``, ``Error``, ``Fatal`` and ``Refactoring``.
+  The 4 digits should not conflict with existing checkers
+  and the first 2 digits should consistent across the checker (except shared messages).
+
+* The ``displayed-message`` is used for displaying the message to the user,
+  once it is emitted.
+
+* The ``message-symbol`` is an alias of the message id
+  and it can be used wherever the message id can be used.
+
+* The ``message-help`` is used when calling ``pylint --help-msg``.
+
+Optionally message can contain optional extra options::
+
+   msgs = {
+       "message-id": (
+           "displayed-message", "message-symbol", "message-help",
+           {"extra_option_name": "extra_option_value"}
+       )
+   }
```

Comment:
I think a full example would work well here, something understandable by everyone like:

```suggestion
* The ``displayed-message`` is used for displaying the message to the user,
  once it is emitted.

* The ``message-symbol`` is an alias of the message id
  and it can be used wherever the message id can be used.

* The ``message-help`` is used when calling ``pylint --help-msg``.

Optionally message can contain optional extra options::

   msgs = {
       "E0401": (
        "Unable to import %s",
        "import-error",
        "Used when pylint has been unable to import a module.",
        {"old_names": [("F0401", "old-import-error")]},
    ),

The message is then formatted using the ``args`` parameter from ``add_message`` i.e. in
``self.add_message("import-error", args=module_we_cant_import, node=importnode)``
    
```